src/contracts/core/AllocationManager.sol

```diff
@@ -200,7 +200,9 @@ contract AllocationManager is
                         // the deallocation delay. This magnitude remains slashable until then.
                         deallocationQueue[operator][strategy].pushBack(operatorSet.key());
 
-                        allocation.effectBlock = uint32(block.number) + DEALLOCATION_DELAY;
+                        // deallocations are slashable in the window [block.number, block.number + deallocationDelay]
+                        // therefore, the effectBlock is set to the block right after the slashable window
+                        allocation.effectBlock = uint32(block.number) + DEALLOCATION_DELAY + 1;
```

Comment:
Regression tests:

1. Assert that a deallocation is not slashable 1 block after the dealloc delay, but is slashable right on it
2. Assert that a registration is not slashable 1 block after the registration but slashable right on it

Reply:
I can add these tmrw as well

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/965/commits/339873cea7ebc156471924a2eb3737e5ff7f6968